akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -116,7 +116,7 @@ private[akka] trait DeathWatch { this: ActorCell =>
   @silent
   private def removeFromMap[T](subject: ActorRef, map: Map[ActorRef, T]): Map[ActorRef, T] =
     if (subject.path.uid != ActorCell.undefinedUid) (map - subject) - new UndefinedUidActorRef(subject)
-    else map.filterKeys(_.path != subject.path).toMap
+    else map.filter { case (k, _) => k.path != subject.path }
```

Comment:
👍 Nice. I guess we could have added a minor back port of https://github.com/akka/akka/pull/26831/files#diff-d52fdc08563814c25f721051d7c238c6L114-L119 when it was removed for 2.5. Thanks for contributing this.